composer/datasets/streaming/download.py

```diff
@@ -24,19 +24,48 @@ def download_from_s3(remote: str, local: str, timeout: float) -> None:
         import boto3
         from botocore.config import Config
     except ImportError as e:
-        raise ImportError(
-            textwrap.dedent("""\
-            Composer was installed without streaming support. To use streaming with Composer, run: `pip install mosaicml
-            [streaming]` if using pip or `conda install -c conda-forge monai` if using Anaconda""")) from e
+        raise MissingConditionalImportError(extra_deps_group="streaming", conda_package="boto3") from e
+
 
     obj = urlparse(remote)
     if obj.scheme != 's3':
         raise ValueError(f"Expected obj.scheme to be 's3', got {obj.scheme} for remote={remote}")
 
     config = Config(read_timeout=timeout)
     s3 = boto3.client('s3', config=config)
-    s3.download_file(obj.netloc, obj.path[1:], local)
+    s3.download_file(obj.netloc, obj.path.lstrip('/'), local)
+
+
+def download_from_sftp(remote: str, local: str) -> None:
+    """Download a file from remote to local.
+    Args:
+        remote (str): Remote path (SFTP).
+        local (str): Local path (local filesystem).
+    """
+    try:
+        from paramiko import RSAKey, SFTPClient, Transport
+    except ImportError as e:
+        raise MissingConditionalImportError(extra_deps_group="streaming", conda_package="paramiko") from e
+
+    obj = urlparse(remote)
+    if obj.scheme != 'sftp':
+        raise ValueError(f"Expected obj.scheme to be 'sftp', got {obj.scheme} for remote={remote}")
+
+    local_tmp = local + ".tmp"
+    if os.path.exists(local_tmp):
+        os.remove(local_tmp)
+    
+    private_key_path = os.environ["STREAMING_DATASET_SSH_FILE"]
+    username = os.environ["STREAMING_DATASET_SSH_USERNAME"]
+    pkey = RSAKey.from_private_key_file(private_key_path)
+    transport = Transport(sock=(obj.netloc, 22))
```

Comment:
do they have any context manager you can  use so `.close` does not have to be called manually?
If not, is it possible to use `with contextlib.closing(Transport(...)) as transport:` and same thing for the `connection` below?